### PR TITLE
log: remove `env_logger` feature and usage

### PR DESCRIPTION
**Pull request by davidbarsky**
Removing the `env_logger` feature in order to address GHSA-g98v-hv3f-hcfr.

For additional details on the approach, please refer to #2750. Note that this
PR depends on #2770, so this PR will temporarily have more commits 
than intended.

### CODE REVIEW

File: tracing-log/src/lib.rs [hunk 1]
```diff
@@ -72,8 +70,6 @@
 //! * `trace-logger`: enables an experimental `log` subscriber, deprecated since
 //!   version 0.1.1.
```
Comment (hawkw) [inline tracing-log/src/lib.rs:71]: let's also remove this before shipping an 0.2.0? i'm fine with doing that in this PR, or in a separate one, up to you.

**Review by hawkw — COMMENTED** [review]
let's also remove the deprecated `log-tracer` module. it's probably easier to do that in this PR?

**Comment by davidbarsky** [comment]
> let's also remove the deprecated `log-tracer` module. it's probably easier to do that in this PR?

Oh, you're right. Let me get on that.

File: tracing-log/src/lib.rs (hunk 1 above)
Comment (davidbarsky) [reply tracing-log/src/lib.rs:71]: done in this PR!

**Review by hawkw — APPROVED** [review]

File: tracing-log/src/lib.rs [hunk 2]
```diff
@@ -143,10 +138,6 @@ use tracing_core::{
 #[cfg_attr(docsrs, doc(cfg(feature = "log-tracer")))]
 pub mod log_tracer;
 
-#[cfg(feature = "trace-logger")]
-#[cfg_attr(docsrs, doc(cfg(feature = "trace-logger")))]
```
Comment (hawkw) [inline tracing-log/src/lib.rs:147]: since this module had a feature flag, can we also remove the feature flag from the `Cargo.toml`?

Comment (davidbarsky) [reply tracing-log/src/lib.rs:147]: whoops, sorry—I had completely missed that.

**Pull request merged**